metadata_mapper/utilities.py

```diff
@@ -0,0 +1,138 @@
+    else:
+        s3 = boto3.resource('s3')
+        bucket = 'rikolti'
+        key = f"{directory}/{collection_id}"
+        return s3.list_objects_v2(bucket, key)
+
+
+def read_from_bucket(directory: str, collection_id: int, file_name: Union[str, int]) -> str:
+    """
+    Reads the contents of a file from the appropriate content bucket.
+
+    Data comes from local filesystem or S3, depending on ENV vars.
+
+    Parameters:
+        collection_id: str
+            Files are separated into directories by collection_id
+        file_name: Union[str, int]
+            The name of the file to read
+
+    Returns: str
+        The file contents
+    """
+    if settings.DATA_SRC == 'local':
+        page_path = os.sep.join([
+            settings.local_path(directory, collection_id),
+            str(file_name)
+        ])
+
+        with open(page_path, "r") as metadata_file:
+            return metadata_file.read()
+    elif settings.DATA_SRC == 's3':
+        s3 = boto3.resource('s3')
+        bucket = 'rikolti'
+        key = f"{directory}/{collection_id}/{file_name}"
+        s3_obj_summary = s3.Object(bucket, key).get()
+        return s3_obj_summary['Body'].read()
+
+
+def read_mapped_metadata(collection_id: int, page_id: int) -> list[dict]:
+    """
+    Reads and parses the content of a mapped metadata file.
+
+    Parameters:
+        collection_id: int
+            The collection ID
+        page_id: int
+            The page ID (filename) to read and parse
+
+    Returns: list[dict]
+        The parsed data
+    """
+    return json.loads(read_from_bucket("mapped_metadata", collection_id, page_id))
+
+
+def read_vernacular_metadata(collection_id: int, page_id: int) -> list[dict]:
+    """
+    Reads and parses the content of a vernacular (unmapped) metadata file.
+
+    Parameters:
+        collection_id: int
+            The collection ID
+        page_id: int
+            The page ID (filename) to read and parse
+
+    Returns: list[dict]
+        The parsed data
+    """
+    return json.loads(read_from_bucket("vernacular_metadata", collection_id, page_id))
+
+
+def write_to_bucket(directory: str, collection_id: int,
+                    file_name: Union[str, int], content: str,
+                    append: bool = False) -> None:
+    if isinstance(content, list) or isinstance(content, dict):
+        content = json.dumps(content)
+
+    if settings.DATA_SRC == 'local':
+        dir_path = settings.local_path(directory, collection_id)
+        if not os.path.exists(dir_path):
+            os.makedirs(dir_path)
+        page_path = os.sep.join([dir_path, str(file_name)])
+
+        with open(page_path, "a" if append else "w") as file:
+            file.write(content)
+    elif settings.DATA_SRC == 's3':
+        s3 = boto3.resource('s3')
+        bucket = 'rikolti'
+        key = f"{directory}/{collection_id}/{file_name}"
+        s3_obj = s3.Object(bucket, key)
+        s3_obj.put(Body=content)
```

Comment:
🎉🙏💯💪 love this utilities module, can't wait to incorporate it into the rest of the metadata_mapper module (and maybe even make it a rikolti-wide utilities module). thank you!